src/runtime/app-login-manager.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2020 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {tryParseJson} from "../utils/json";
+import {UserState} from '../model/user-state';
+
+const LOGIN_REDIRECT_PARAM = 'googleLoginRedirect';
+
+/**
+ */
+export class AppLoginManager {
```

Comment:
Can we add some logging to this to help with debugging?

Reply:
Done.